proposals/0052-svg-components.md

```diff
@@ -0,0 +1,328 @@
+- Stage 2 Issue: https://github.com/withastro/roadmap/issues/699
+- Stage 3 PR: https://github.com/withastro/roadmap/pull/1035
+
+# Summary
+
+This RFC proposes adding native support for importing and rendering `.svg` files as Astro components, with optimized rendering techniques to minimize performance impact. It aims to allow `.svg` files to be treated as components that accept props and be optimized for repeated use on a page using `<symbol>` and `<use>` elements.
+
+# Example
+
+An SVG can be imported directly into an Astro or MDX file and treat the default export as an Astro component.
+
+```astro
+---
+import Logo from '../assets/logo.svg'
+---
+
+<Logo />
+```
+
+# Background & Motivation
+
+Currently, using `.svg` files in Astro requires workarounds that can result in unnecessary complexity or bloated client-side JavaScript. As `.svg` files are widely used for icons and graphics, an optimized solution is necssary to simplify usage and improve performance.
+
+Previously, `astro-icon` was introduced to experiment with `.svg` support but, with the relase of `astro:assets`, there's now an opportunity to provide optimized `.svg` imports that simplify the developer experience and reduce client-side bloat.
+
+The core motivation is to automate best practices around `.svg` usage, enabled developers to effortlessly use and optimize their `.svg` assets in Astro projects.
+
+# Goals
+
+- Enable importing and rendering `.svg` files as Astro components.
+- Enable importing and rendering `.svg` files in MDX files
+- Allow users to pass props to the root `svg` element while allowing for prop overrides.
+- Inline `.svg` file directly without using optimized sprite rendering behavior.
+- Allow users to opt-in to optimized `.svg` rendering by utilizing `symbol` and `use` for repeated instances.
+- Maintain backwards compatibility with existing `.svg` import behavior.
+- Strip unnecessary attributes like `xmlns` and `version` from `.svg` files to reduce size.
+- Allow third-party plugins to hook into the `.svg` import behavior to generate their own component libraries.
+
+# Non-Goals
+
+- Out-of-scope: Advanced tree-shaking or SSR optimizations for large icon sets.
+- Out-of-scope: Advanced `.svg` file optimization such as `svgo`.
+- Future: Adding support for `.svg` files as framework components.
+- Non-goal: Adding support for `.json` icon sets such as `@iconify/json`.
+- Non-goal: SVG content analysis for accessibility.
+
+# Detailed Design
+
+## Importing SVG Files as Components
+
+The core design of this proposal allows `.svg` files to be imported and used as components in Astro. This means developers can import `.svg` files into their Astro components and render them directly as if they were Astro components. The goal is to maintain ease of use while automating performance optimizations under the hood.
+
+**Example:**
+
+```astro
+---
+import Logo from '../assets/logo.svg';
+---
+
+<Logo />
```

Comment:
Is it possible to have svg work like the image component, because I lot of imports would be a waste of time and take website resources. Where you import the component and then provide the svgs inside of the component.